<a href="https://colab.research.google.com/github/mohansaidinesh/Machine-Learning/blob/main/Experiment5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

###Experiment : 5

Build a Logistic Regression classifier by considering a suitable dataset.

In [ ]:
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import classification_report, confusion_matrix

In [ ]:
#loading the dataset
data=pd.read_csv("https://assets.datacamp.com/production/repositories/628/datasets/444cdbf175d5fbf564b564bd36ac21740627a834/diabetes.csv")
data.head()

pregnancies  glucose  diastolic  triceps  insulin   bmi    dpf  age  \
0            6      148         72       35        0  33.6  0.627   50   
1            1       85         66       29        0  26.6  0.351   31   
2            8      183         64        0        0  23.3  0.672   32   
3            1       89         66       23       94  28.1  0.167   21   
4            0      137         40       35      168  43.1  2.288   33   

   diabetes  
0         1  
1         0  
2         1  
3         0  
4         1

In [ ]:
X = data.drop('diabetes', axis=1).values
y = data['diabetes'].values

In [ ]:
#split the dataset into training and testing data
X_train, X_test, y_train, y_test =train_test_split(X, y, test_size=0.1, random_state=42)

In [ ]:
#normalization of data
scaler = StandardScaler()
X_train = scaler.fit_transform(X_train)

fitting the model

In [ ]:
model = LogisticRegression(C=0.01,random_state=42)
model.fit(X_train, y_train)

LogisticRegression(C=0.01, random_state=42)

In [ ]:
#normalization of testing data
X_test = scaler.transform(X_test)
y_pred = model.predict(X_test)

In [ ]:
model.score(X_train, y_train)

0.7713458755426917

In [ ]:
model.score(X_test, y_test)

0.7662337662337663

confusion matrix for evaluation of model

In [ ]:
confusion_matrix(y_test, y_pred)

array([[44,  6],
       [12, 15]])

In [ ]:
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       0.79      0.88      0.83        50
           1       0.71      0.56      0.63        27

    accuracy                           0.77        77
   macro avg       0.75      0.72      0.73        77
weighted avg       0.76      0.77      0.76        77



**CONCLUSION :** The accuracy of the training data is 0.77 and for the testing data is 0.76.